vyper/semantics/analysis/base.py

```diff
@@ -234,6 +234,21 @@ class VarAccess:
     # A sentinel indicating a subscript access
     SUBSCRIPT_ACCESS: ClassVar[Any] = object()
 
+    # custom __reduce__ and _produce implementations to work around
+    # a pickle bug.
+    # see https://github.com/python/cpython/issues/124937#issuecomment-2392227290
+    def __reduce__(self):
+        dict_obj = {f.name: getattr(self, f.name) for f in fields(self)}
+        return self.__class__._produce, (dict_obj,)
```

Comment:
You could also use the following trick if all fields can be passed as keyword arguments to constructor in subclasses:
```suggestion
        return partial(self.__class__, **dict_obj), ()
```

Reply:
interesting, is there a performance difference?

Reply:
It may be slower. The advantage is that you do not need to add private method `_produce()`.